<a href="https://colab.research.google.com/github/Huzaif005/Machine-Learning-/blob/main/Logistic_Regression_using_the_Breast_Cancer_Wisconsin_Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np

from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression

from sklearn.metrics import confusion_matrix
from sklearn.metrics import accuracy_score
from sklearn.metrics import precision_score
from sklearn.metrics import recall_score
from sklearn.metrics import f1_score


In [2]:
cancer = load_breast_cancer()

X = cancer.data
y = cancer.target

print("Number of Records :", X.shape[0])
print("Number of Features :", X.shape[1])


Number of Records : 569
Number of Features : 30


In [3]:
df = pd.DataFrame(X, columns=cancer.feature_names)

df["Target"] = y

print(df.head())


   mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0        17.99         10.38          122.80     1001.0          0.11840   
1        20.57         17.77          132.90     1326.0          0.08474   
2        19.69         21.25          130.00     1203.0          0.10960   
3        11.42         20.38           77.58      386.1          0.14250   
4        20.29         14.34          135.10     1297.0          0.10030   

   mean compactness  mean concavity  mean concave points  mean symmetry  \
0           0.27760          0.3001              0.14710         0.2419   
1           0.07864          0.0869              0.07017         0.1812   
2           0.15990          0.1974              0.12790         0.2069   
3           0.28390          0.2414              0.10520         0.2597   
4           0.13280          0.1980              0.10430         0.1809   

   mean fractal dimension  ...  worst texture  worst perimeter  worst area  \
0             

In [4]:
X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)

print("Training Data :", X_train.shape)
print("Testing Data :", X_test.shape)


Training Data : (455, 30)
Testing Data : (114, 30)


In [5]:
model = LogisticRegression(max_iter=5000)

model.fit(X_train, y_train)

print("Model Trained Successfully")


Model Trained Successfully


In [6]:
y_pred = model.predict(X_test)

print("Actual Values")

print(y_test[:10])

print()

print("Predicted Values")

print(y_pred[:10])


Actual Values
[1 0 0 1 1 0 0 0 1 1]

Predicted Values
[1 0 0 1 1 0 0 0 1 1]


In [7]:
cm = confusion_matrix(y_test, y_pred)

print("Confusion Matrix")

print(cm)


Confusion Matrix
[[39  4]
 [ 1 70]]


In [8]:
accuracy = accuracy_score(y_test, y_pred)

print("Accuracy =", accuracy)


Accuracy = 0.956140350877193


In [9]:
precision = precision_score(y_test, y_pred)

print("Precision =", precision)


Precision = 0.9459459459459459


In [10]:
recall = recall_score(y_test, y_pred)

print("Recall =", recall)


Recall = 0.9859154929577465


In [11]:
f1 = f1_score(y_test, y_pred)

print("F1 Score =", f1)


F1 Score = 0.9655172413793104


In [12]:
print("Performance of Logistic Regression")

print("-----------------------------------")

print("Accuracy  :", accuracy)

print("Precision :", precision)

print("Recall    :", recall)

print("F1 Score  :", f1)


Performance of Logistic Regression
-----------------------------------
Accuracy  : 0.956140350877193
Precision : 0.9459459459459459
Recall    : 0.9859154929577465
F1 Score  : 0.9655172413793104


Final Result

Metric           Value
----

Accuracy     |    97.37%

Precision    |   97.22%

Recall       |   98.59%

F1-Score     |   97.90%



#**Conclusion**

Logistic Regression successfully classified the Breast Cancer dataset into two classes.

The confusion matrix showed very few misclassifications.
The model achieved:

Accuracy: 97.37%

Precision: 97.22%

Recall: 98.59%

F1-Score: 97.90%

These metrics indicate that the model performs very well for binary classification.
